docs/getting-started/basics.md

```diff
@@ -0,0 +1,389 @@
+
+You cannot do an `if` statement on vars in components, since the value is not known at compile time.
+
+```python
+class BadState(rx.State):
+    count: int = 0
+    
+def count_if_even():
+    return rx.box(
+        rx.heading("Count: "),
+        # This will raise a compile error, as MyState.count is a var and not known a compile time.
+        rx.text(BadState.count if BadState.count % 2 == 0 else "Odd"),
+        # Using an if statement with a var as a prop will NOT work either.
+        rx.text("hello", color="red" if BadState.count % 2 == 0 else "blue"),
+    )
+```
+
+You cannot do a `for` loop over a list of vars.
+
+```python
+class BadState(rx.State):
+    items: list[str] = ["Apple", "Banana", "Cherry"]
+
+def loop_over_list():
+    return rx.box(
+        # This will raise a compile error, as MyState.items is a list and not known at compile time.
+        *[rx.text(item) for item in BadState.items]
+    )
+```
+
+You cannot do arbitrary Python operations on state vars in components.
+
+```python
+class BadTextState(rx.State):
+    text: str = "Hello world"
+
+def format_text():
+    return rx.box(
+        # Python operations such as `len` will not work on state vars.
+        rx.text(len(BadTextState.text)),
+    )
+```
+
+In the next sections, we will show how to handle these cases.
+
+## Conditional rendering
+
+As mentioned above, you cannot use Python `if/else` statements with state vars in components. Instead, use the [`rx.cond`]({docs.components.conditional_rendering.path}) function to conditionally render components.
+
+```python demo exec
+class LoginState(rx.State):
+    logged_in: bool = False
+
+    def toggle_login(self):
+        self.logged_in = not self.logged_in
+
+def show_login():
+    return rx.box(
+        rx.cond(
+            LoginState.logged_in,
+            rx.heading("Logged In"),
+            rx.heading("Not Logged In"),
+        ),
+        rx.button("Toggle Login", on_click=LoginState.toggle_login)
+    )
+```
+
+## Rendering lists
+
+To iterate over a var that is a list, use the [`rx.foreach`]({docs.components.rendering_iterables.path}) function to render a list of components.
+
+Pass the list var and a function that returns a component as arguments to `rx.foreach`.
+
+```python demo exec
+class ListState(rx.State):
+    items: list[str] = ["Apple", "Banana", "Cherry"]
+
+def render_item(item: rx.Var[str]):
+    """Render a single item."""
+    # Note that item here is a Var, not a str!
+    return rx.list.item(item)
+
+def show_fruits():
+    return rx.box(
+        rx.foreach(ListState.items, render_item),
+    )
+```
+
+The function that renders each item takes in a `Var`, since this will get compiled up front.
+
+## Var Operations
+
+You can't use arbitrary Python operations on state vars in components, but Reflex has [var operations]({docs.vars.var_operations.path}) that you can use to manipulate state vars.
+
+For example, to check if a var is even, use the `is_even` var operation.
```

Comment:
where is `is_even`? looks like this is just inline modular arithmetic and equality

Reply:
copilot leftover, removed